pkg/resources/istioingress/istioingress.go

```diff
@@ -59,9 +62,11 @@ func New(client client.Client, cluster *v1beta1.KafkaCluster) *Reconciler {
 // Reconcile implements the reconcile logic for IstioIngress
 func (r *Reconciler) Reconcile(log logr.Logger) error {
 	log = log.WithValues("component", componentName)
-
 	log.V(1).Info("Reconciling")
 	if r.KafkaCluster.Spec.GetIngressController() == istioingress.IngressControllerName {
+		if r.KafkaCluster.Spec.IstioControlPlane == nil {
```

Comment:
We should check if `istioControlPlane` is populated only if there is at least one external listener that uses ingress controller.